src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Completion/CompletionService.cs

```diff
@@ -0,0 +1,166 @@
+
+                    // wait for the queue item
+                    queueItem.ItemProcessed.WaitOne();
+                    var completionResult = queueItem.GetResultAsT<AutoCompletionResult>();
+                    if (completionResult != null && completionResult.CompletionItems != null)
+                    {
+                        result = completionResult;
+                    }
+                    else if (!ShouldShowCompletionList(scriptDocumentInfo.Token))
+                    {
+                        result.CompleteResult(scriptDocumentInfo.ScriptParseInfo.CurrentSuggestions, AutoCompleteHelper.EmptyCompletionList);
+                    }
+                }
+                finally
+                {
+                    Monitor.Exit(scriptDocumentInfo.ScriptParseInfo.BuildingMetadataLock);
+                }
+            }
+
+            return result;
+        }
+
+        private QueueItem AddToQueue(
+            ConnectionInfo connInfo,
+            ScriptParseInfo scriptParseInfo,
+            ScriptDocumentInfo scriptDocumentInfo,
+            bool useLowerCaseSuggestions)
+        {
+            // queue the completion task with the binding queue    
+            QueueItem queueItem = this.BindingQueue.QueueBindingOperation(
+                key: scriptParseInfo.ConnectionKey,
+                bindingTimeout: LanguageService.BindingTimeout,
+                bindOperation: (bindingContext, cancelToken) =>
+                {
+                    return CreateCompletionsFromSqlParser(connInfo, scriptParseInfo, scriptDocumentInfo, bindingContext.MetadataDisplayInfoProvider);
+                },
+                timeoutOperation: (bindingContext) =>
+                {
+                    // return the default list if the connected bind fails
+                    return CreateDefaultCompletionItems(scriptParseInfo, scriptDocumentInfo, useLowerCaseSuggestions);
+                });
+            return queueItem;
+        }
+
+        private static bool ShouldShowCompletionList(Token token)
+        {
+            bool result = true;
+            if (token != null)
+            {
+                switch (token.Id)
+                {
+                    case (int)Tokens.LEX_MULTILINE_COMMENT:
+                    case (int)Tokens.LEX_END_OF_LINE_COMMENT:
+                        result = false;
+                        break;
+                }
+            }
+            return result;
+        }
+
+        private ConnectedBindingQueue BindingQueue { get; set; }
+
+        private AutoCompletionResult CreateDefaultCompletionItems(ScriptParseInfo scriptParseInfo, ScriptDocumentInfo scriptDocumentInfo, bool useLowerCaseSuggestions)
+        {
+            AutoCompletionResult result = new AutoCompletionResult();
+            CompletionItem[] completionList = AutoCompleteHelper.GetDefaultCompletionItems(scriptDocumentInfo, useLowerCaseSuggestions);
```

Comment:
>AutoCompleteHelper [](start = 46, length = 18)

I think the refactor starts to mix-up responsibilities between the AutoCompleteHelper and the CompletionService..i.e., not really sure what should go in which class.  It's probably fine for this check-in, but we should rationalize why we have to similar classes and when to use which one, or should we get rid of the helper and move that functionality into here.